explorer/testdata/match/fail_exhaustive_int.carbon

```diff
@@ -0,0 +1,23 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: not %{explorer} %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes=false %s
+// RUN: not %{explorer} --parser_debug --trace_file=- %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes %s
+// AUTOUPDATE: %{explorer} %s
+
+package ExplorerTest api;
+
+fn A(n: i32) -> i32 {
+  match (n) {
+    case 1 => { return 1; }
+    case 2 => { return 2; }
+  }
+// CHECK: COMPILATION ERROR: {{.*}}/explorer/testdata/match/fail_exhaustive_int.carbon:[[@LINE+1]]: non-exhaustive match may allow control-flow to reach the end of a function that provides a `->` return type
```

Comment:
I admit, think makes me want a couple tests that should pass:

```
fn A(n: i32) -> i32 {
  match (n) {
    case 1 => { return 1; }
    case 2 => { return 2; }
  }
  return 3;
}
```

```
fn A(n: i32) {
  match (n) {
    case 1 => { return; }
    case 2 => { return; }
  }
}
```

Reply:
Sure, sounds reasonable. I think these are tests for the "control reaches end of function" diagnostic, not tests for the pattern match exhaustiveness checking code, so I've added them to testdata/function/.